src/Zypper.cc

```diff
@@ -340,22 +340,39 @@ namespace {
     return mayuse;
   }
 
-  inline std::string legacyCLIStr( const std::string & old_r, const std::string & new_r, bool global_r = false )
-  {
-    return str::FormatNAC( global_r
-			 ? _("Legacy commandline option %1% detected. Please use global option %2% instead.")
-			 : _("Legacy commandline option %1% detected. Please use %2% instead.") )
-			 % NEGATIVEString(dashdash(old_r))
-			 % POSITIVEString(dashdash(new_r));
+  enum LegacyCLIMsgType {
+    Local,
+    Global,
+    Ignored
+  };
+
+  inline std::string legacyCLIStr( const std::string & old_r, const std::string & new_r, LegacyCLIMsgType type )
```

Comment:
Throughout the reviews I realize how much I actually miss function args not following the `_r` convention. I'd vote for keeping this as a rule (for new code). So `type_r`.

Throughout my code there are just a few exceptions. Binary operator  arguments are usually just `lhs` and `rhs` and ostream operator<< uses `str` and `obj`. The rest usually follows the `_r` convention.

Reply:
Done